dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageRenderUtil.java

```diff
@@ -228,143 +230,191 @@ private Container getWorkingContainerById(final String containerIdOrPath, final
             }
-            
+
             final List<ContainerStructure> containerStructures = APILocator.getContainerAPI().getContainerStructures(container);
-            final boolean hasWritePermissionOnContainer = permissionAPI.doesUserHavePermission(container, PERMISSION_WRITE, user, false)
-                    && APILocator.getPortletAPI().hasContainerManagerRights(user);
-            final boolean hasReadPermissionOnContainer = permissionAPI.doesUserHavePermission(container, PERMISSION_READ, user, false);
-            contextMap.put("EDIT_CONTAINER_PERMISSION" + container.getIdentifier(), hasWritePermissionOnContainer);
-            if (Config.getBooleanProperty("SIMPLE_PAGE_CONTENT_PERMISSIONING", true)) {
-                contextMap.put("USE_CONTAINER_PERMISSION" + container.getIdentifier(), true);
-            } else {
-                contextMap.put("USE_CONTAINER_PERMISSION" + container.getIdentifier(), hasReadPermissionOnContainer);
-            }
+            this.addPermissions(container);
 
-            final Map<String, List<Map<String,Object>>> contentMaps = Maps.newLinkedHashMap();
-            final Map<String, List<String>> containerUuidPersona = Maps.newHashMap();
+            final Map<String, List<Map<String, Object>>> contentMaps = Maps.newLinkedHashMap();
+            final Map<String, List<String>> containerUuidPersona     = Maps.newHashMap();
             for (final String uniqueId : pageContents.row(containerId).keySet()) {
 
                 final String uniqueUUIDForRender = needParseContainerPrefix(container, uniqueId) ?
-                                ParseContainer.getDotParserContainerUUID(uniqueId) :
-                                uniqueId;
+                        ParseContainer.getDotParserContainerUUID(uniqueId) : uniqueId;
 
-                if(ContainerUUID.UUID_DEFAULT_VALUE.equals(uniqueId)) {
+                if (ContainerUUID.UUID_DEFAULT_VALUE.equals(uniqueId)) {
                     continue;
                 }
 
                 final Collection<PersonalizedContentlet> personalizedContentletSet = pageContents.get(containerId, uniqueId);
-                final List<Map<String, Object>> personalizedContentletMap = Lists.newArrayList();
+                final List<Map<String, Object>> personalizedContentletMap          = Lists.newArrayList();
+                int   contentletIncludedCount = 1;
+                for (final PersonalizedContentlet personalizedContentlet : personalizedContentletSet) {
 
-                for (final PersonalizedContentlet  personalizedContentlet : personalizedContentletSet) {
-                    final Contentlet contentlet = getContentlet(personalizedContentlet);
+                    final Contentlet contentlet = this.getContentlet(personalizedContentlet);
 
                     if (contentlet == null) {
+
+                        continue;
+                    }
+
+                    if (container.getMaxContentlets() < contentletIncludedCount++) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid assignments in operands](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5316867)